kratos/geometries/geometry.h

```diff
@@ -733,6 +734,54 @@ class Geometry
     ///@name Inquiry
     ///@{
 
+    /**
+     * @brief Checks if two GeometryType have the same type
+     * @return True if the objects are the same type, false otherwise
+     */
+    inline static bool HasSameType(const GeometryType& rLHS, const GeometryType& rRHS) {
+        return (typeid(rLHS) == typeid(rRHS));
+    }
+
+    /**
+     * @brief Checks if two GeometryType have the same type (pointer version)
+     * @return True if the objects are the same type, false otherwise
+     */
+    inline static bool HasSameType(const GeometryType * rLHS, const GeometryType* rRHS) {
+        return GeometryType::HasSameType(*rLHS, *rRHS);
+    }
+
+    /**
+     * @brief Checks if two GeometryType have the same geometry type
+     * @return True if the geometries are the same type, false otherwise
+     */
+    inline static bool HasSameGeometryType(const GeometryType& rLHS, const GeometryType& rRHS) {
```

Comment:
```suggestion
    inline static bool HasSameGeometryType(
        const GeometryType& rLHS,
        const GeometryType& rRHS)
    {
```
(minor style guide comment)